--- ../TraceReader.backup/CMakeLists.txt	2014-07-15 17:25:32.667452910 +0200
+++ CMakeLists.txt	2014-07-15 23:27:57.631343495 +0200
@@ -42,6 +42,7 @@
 
 add_subdirectory(test)
 add_subdirectory(plugin)
+add_subdirectory(feign_siox-provider)
 
 install(TARGETS siox-tools-TraceReader siox-trace-reader LIBRARY DESTINATION ${SIOX_INSTALL_LIBRARY_DIR} RUNTIME DESTINATION bin ARCHIVE DESTINATION ${SIOX_INSTALL_LIBRARY_DIR})
 
