--- ../TraceReader.backup/TraceReader.hpp	2014-07-15 17:25:32.665452949 +0200
+++ TraceReader.hpp	2014-07-16 22:53:44.077162588 +0200
@@ -54,14 +54,16 @@
 				delete(t);
 			}			
 		}
+
+		SystemInformationGlobalIDManager * s = nullptr;
+
 	private:
 		AssociationMapper * a = nullptr;
 		Ontology * o = nullptr;
-		SystemInformationGlobalIDManager * s = nullptr;
 		Topology * t = nullptr;
 		ActivitySerializationPlugin * activityDeserializer;
 
 		void strattribute( const Attribute & a, stringstream & s ) throw( NotFoundError );
 };
 
-#endif
\ No newline at end of file
+#endif
